int32 client_id
string[] shelves
int32[] item_quantities
